### PR TITLE
Add ClaimQueue wrapper

### PR DESCRIPTION
**Pull request by sandreim**
Remove `fetch_next_scheduled_on_core` in favor of new wrapper and methods for accessing it.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5738114

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5738415

**Review by tdimitrov — COMMENTED** [review]
That's a good idea. A few subsystem will work with the claim queue and will need similar boilerplate code to access it.

File: polkadot/node/subsystem-util/src/vstaging.rs [hunk 1]
```diff
@@ -23,14 +23,37 @@ use std::collections::{BTreeMap, VecDeque};
...
+impl ClaimQueueSnapshot {
+	/// Returns the `ParaId` that has a claim for `core_index` at the specified `depth` in the
+	/// claim queue. A depth of `0` means the very next block.
+	pub fn get_claim_for(&self, core_index: CoreIndex, depth: usize) -> Option<ParaId> {
+		self.0.get(&core_index)?.get(depth).copied()
+	}
+
+	/// Returns an iterator over all claimed cores and the claiming `ParaId` at the specified
+	/// `depth` in the claim queue.
+	pub fn iter_claims_at(&self, depth: usize) -> impl Iterator<Item = (CoreIndex, ParaId)> + '_ {
```
Comment (eskimor) [inline polkadot/node/subsystem-util/src/vstaging.rs:51]: Maybe a slightly more verbose `iter_claims_at_depth`? When I looked at the code my initial understanding at the calling side wast that this is per `CoreIndex`.

**Review by eskimor — COMMENTED** [review]
Looks nice!

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5749058

**Review by tdimitrov — APPROVED** [review]

**Review by eskimor — APPROVED** [review]

**Review by alexggh — APPROVED** [review]

**Pull request merged**